build-support/bin/ci.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python3
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import argparse
+import os
+import platform
+import subprocess
+from enum import Enum
+from pathlib import Path
+from typing import List, Optional
+
+from common import banner, die, green, travis_section
+
+
+def main() -> None:
+  banner("CI BEGINS")
+
+  args = create_parser().parse_args()
+  setup_environment(python_version=args.python_version)
+
+  if args.bootstrap:
+    bootstrap(clean=args.bootstrap_clean, python_version=args.python_version)
+  set_run_from_pex()
+
+  if args.precommit_checks:
+    run_precommit_checks()
+  if args.sanity_checks:
+    run_sanity_checks()
+  if args.lint:
+    run_lint()
+  if args.doc_gen:
+    run_doc_gen_tests()
+  if args.clippy:
+    run_clippy()
+  if args.cargo_audit:
+    run_cargo_audit()
+  if args.python_tests:
+    run_python_tests()
+  if args.rust_tests:
+    run_rust_tests()
+  if args.jvm_tests:
+    run_jvm_tests()
+  if args.integration_tests:
+    run_integration_tests(shard=args.integration_shard)
+  if args.internal_backend_tests:
+    run_internal_backend_tests()
+  if args.platform_specific_tests:
+    run_platform_specific_tests()
+
+  banner("CI ENDS")
+  print()
+  green("SUCCESS")
+
+# -------------------------------------------------------------------------
+# Options
+# -------------------------------------------------------------------------
+
+class PythonVersion(Enum):
+  py27 = "2.7"
+  py36 = "3.6"
+  py37 = "3.7"
+
+  def __str__(self) -> str:
+    return str(self.value)
+
+
+def create_parser() -> argparse.ArgumentParser:
+  # TODO: reassign the short CLI flag names to reflect the long names.
```

Comment:
Is there any reason to keep the short cli flag names at all? If so is there any reason to keep the old values? Since ci.sh / ci.py is script / alias / habit breaking already and the audience for running ci scripts is small, it seems to me safe to dio all the breaking right now. It certainly doesn't harm readability of the diff.

Reply:
Great.

I definitely don't buy the argument for keeping short flags but this is your call. Short flags are useful for personal runs of tools but this "tool" is only ever used to repro a CI failure, in which case you want to use the options used in CI to actually repro or else ... that's it I think. I may be missing something here. Do _you_ actually switch to using short options for local development? You're really the only local developer here.

Reply:
That's fair! I'll kill off the short names. 

One major benefit of doing so is that it becomes easier to introduce new flags, without having to worry that the short letter you wanted is already taken.

Thanks for the suggestions!